### PR TITLE
Rust API article

### PR DESCRIPTION
**Pull request by chenkie**
screencap

### CODE REVIEW

**Comment by chenkie** [comment]
SEO Checklist: https://docs.google.com/a/auth10.com/document/d/1FpM3s-J5Tqz5NyPtAFswMCsoFvslNqi6IrzVKlONrzA/edit?usp=sharing

File: _posts/2015-11-26-build-an-api-in-rust-with-jwt-authentication-using-nickelrs [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+- mongodb
+- api
+- rest
+- nickel.rs
+- frank_jwt
+- hyper
+---
+
+---
+**TL;DR:** Open source crates like **[Nickel.rs](http://nickel.rs/)** and the **[MongoDB Rust Driver](https://github.com/mongodb-labs/mongo-rust-driver-prototype)** make it possible to create RESTful APIs in the Rust language. In this article, we cover how to do `GET`, `POST` and `DELETE` requests on user data. Check out the [repo](https://github.com/auth0/rust-api-example) to get the code.
```
Comment (mgonto) [inline _posts/2015-11-26-build-an-api-in-rust-with-jwt-authentication-using-nickelrs]: Crates? I don't get it :D

Comment (mgonto) [inline _posts/2015-11-26-build-an-api-in-rust-with-jwt-authentication-using-nickelrs]: Shouldn't we link to the simple interceptor you did here as well?

File: _posts/2015-11-26-build-an-api-in-rust-with-jwt-authentication-using-nickelrs.markdown [hunk 2]
```diff
@@ -0,0 +1,536 @@
...
+- mongodb
+- api
+- rest
+- nickel.rs
+- frank_jwt
+- hyper
+---
+
+---
+**TL;DR:** Open source crates like **[Nickel.rs](http://nickel.rs/)** and the **[MongoDB Rust Driver](https://github.com/mongodb-labs/mongo-rust-driver-prototype)** make it possible to create RESTful APIs in the Rust language. In this article, we cover how to do `GET`, `POST` and `DELETE` requests on user data. Check out the [repo](https://github.com/auth0/rust-api-example) to get the code.
```
Comment (mgonto) [inline _posts/2015-11-26-build-an-api-in-rust-with-jwt-authentication-using-nickelrs.markdown]: Why not mention our simple library as well?

File: _posts/2015-11-26-build-an-api-in-rust-with-jwt-authentication-using-nickelrs.markdown [hunk 3]
```diff
@@ -0,0 +1,536 @@
...
+
+## Installing Rust and Setting Up the Project
+
+To create and serve our API, we'll use **[Nickel.rs](http://nickel.rs/)**, and to interact with the database, we'll use the **[MongoDB Rust Driver](https://github.com/mongodb-labs/mongo-rust-driver-prototype)**. There are other crates (aka packages) available, especially for creating a server and API, but Nickel.rs offers an abstraction that provides a similar feel to NodeJS and, in particular, Express. This can be helpful for those coming from a Node background. 
+
+If you don't already have Rust installed, the easiest way to get it is with the **rustup** script. If you're not on a Mac, or if you want to build from source, see the [full installation instructions](https://doc.rust-lang.org/stable/book/installing-rust.html).
+
+```bash
+curl -sf -L https://static.rust-lang.org/rustup.sh | sh
```
Comment (mgonto) [inline _posts/2015-11-26-build-an-api-in-rust-with-jwt-authentication-using-nickelrs.markdown]: Isn't this TOO basic? to tell people how to install rust?

File: _posts/2015-11-26-build-an-api-in-rust-with-jwt-authentication-using-nickelrs.markdown [hunk 4]
```diff
@@ -0,0 +1,536 @@
...
+
+```rust
+// src/main.rs
+
+...
+
+fn authenticator<'mw>(request: &mut Request, response: Response<'mw>, ) -> MiddlewareResult<'mw> {
+
+  // We don't want to apply the middleware to the login route
+  if request.origin.uri.to_string() == "/login".to_string() {
```
Comment (mgonto) [inline _posts/2015-11-26-build-an-api-in-rust-with-jwt-authentication-using-nickelrs.markdown]: This middleware shouldn't be used for OPTIONS request.

Isn't it worth open sourcing this middleware separated to the post itself?

File: _posts/2015-11-26-build-an-api-in-rust-with-jwt-authentication-using-nickelrs.markdown [hunk 5]
```diff
@@ -0,0 +1,536 @@
...
+
+We can test everything out by including the JWT as a header when making a request to `/users`.
+
+![rust api jwt GET request](https://cdn.auth0.com/blog/rust-api/rust-api-6.png)
+
+> **Note:** This implementation doesn't deal with JWT expiry, which would be necessary for security in a production app.
+
+## Aside: Authentication is Easy with Auth0
+
+Auth0 issues [JSON Web Tokens](http://jwt.io) on every login for your users. This means that you can have a solid [identity infrastructure](https://auth0.com/docs/identityproviders), including [single sign-on](https://auth0.com/docs/sso/single-sign-on), user management, support for social (Facebook, Github, Twitter, etc.), enterprise (Active Directory, LDAP, SAML, etc.) and your own database of users with just a few lines of code. Auth0 is perfect for [Single Page Applications](https://auth0.com/docs/sequence-diagrams) and very easy to set up.
```
Comment (mgonto) [inline _posts/2015-11-26-build-an-api-in-rust-with-jwt-authentication-using-nickelrs.markdown]: Don't use the generic aside anymore :D. 

In here, show how you can configure the JWT middleware from above to actually work with your Auth0 account secret and tell everybody that they can skip implementing the `/login`

**Pull request merged**